pkg/virt-handler/virtwrap/manager.go

```diff
@@ -95,23 +96,21 @@ type LibvirtDomainManager struct {
 
 func NewConnection(uri string, user string, pass string) (Connection, error) {
 	virConn, err := newConnection(uri, user, pass)
-	if err != nil {
-		return nil, err
-	}
-	return &LibvirtConnection{VirConnection: virConn, user: user, pass: pass, uri: uri, alive: true}, nil
+	lvConn := &LibvirtConnection{Connect: *virConn, user: user, pass: pass, uri: uri, alive: true}
+	return lvConn, err
+
 }
 
-func newConnection(uri string, user string, pass string) (libvirt.VirConnection, error) {
-	var virConn libvirt.VirConnection
+func newConnection(uri string, user string /**pass*/, _ string) (*libvirt.Connect, error) {
+	var virConn *libvirt.Connect
 	var err error
 	if user == "" {
-		virConn, err = libvirt.NewVirConnection(uri)
-	} else {
-		virConn, err = libvirt.NewVirConnectionWithAuth(uri, user, pass)
-	}
-	if err != nil {
-		return libvirt.VirConnection{}, err
+		virConn, err = libvirt.NewConnect(uri)
 	}
+	// TODO: Reimplement using the SASL style ConnectAuth
```

Comment:
This test does exactly what you need:

https://github.com/libvirt/libvirt-go/blob/33d176798e9e86e3d2253ade57e24bb85a885a68/integration_test.go#L232